### PR TITLE
Update exception msg for circular dependency

### LINKED ISSUE #3147: Better exception message for circular dependency in FK cycle
Hi, 
Working with the new dnx-clr-win-x86.1.0.0-Beta7.

I'm getting the following message : 
InvalidOperationException: A circular dependency was detected: 'Schema' {'FutureId'} -> 'Occurrence' {'ID'}, 'Occurrence' {'SchemaID'} -> 'Schema' {'ID'}, 'OccurrenceHistory' {'OccurrenceID'} -> 'Occurrence' {'ID'}.

Where classes architecture is :
- Schema holds all Occurrences under "OccurrenceHistory" enumerable.
- Schema holds Future instance of the occurrence.
- Schema holds Past instance of that occurrence.
(that is, future is the next link in the chain, and past is the previous link in the chain. the history refers to all links whatsoever)

So, i'm not so sure why i get this error, it should work (unless i'm missing something), 
it worked on Beta6 but had some other issues, where it wasn't able to load the Future item with .Include.

Thanks,
Eliran.

### PR DESCRIPTION
**Pull request by smitpatel**
resolves #3147 
@rowanmiller

### CODE REVIEW

**Comment by smitpatel** [comment]
Working on correcting the cycle.

**Comment by rowanmiller** [comment]
:shipit:

**Comment by smitpatel** [comment]
Updated cycle detection to remove tail of a cycle.

File: src/EntityFramework.Core/Internal/Multigraph.cs [hunk 1]
```diff
@@ -292,18 +292,34 @@ public virtual IReadOnlyList<List<TVertex>> BatchingTopologicalSort()
                 }
                 cycle.Reverse();
```
Comment (AndriySvyryd) [inline src/EntityFramework.Core/Internal/Multigraph.cs]: You can rename `cycle` to `cyclicWalk` and `lastCycle` to `cycle`

**Comment by AndriySvyryd** [comment]
:shipit:

File: src/EntityFramework.Core/Internal/Multigraph.cs (hunk 1 above)
Comment (smitpatel) [inline src/EntityFramework.Core/Internal/Multigraph.cs]: :+1:

**Comment by mikary** [comment]
I think the best option for this would be to change the interpretation of the `formatCycle` lambda. Right now it's expected to produce a string that is just the formatted data portion of the message, we could have it return the entire error message instead to support custom error messages like this one.

**Pull request merged**